journal/journal.go

```diff
@@ -98,12 +104,7 @@ func Send(message string, priority Priority, vars map[string]string) error {
 
 		rights := syscall.UnixRights(int(file.Fd()))
 
-		/* this connection should always be a UnixConn, but better safe than sorry */
-		unixConn, ok := conn.(*net.UnixConn)
-		if !ok {
-			return journalError("can't send file through non-Unix connection")
-		}
-		unixConn.WriteMsgUnix([]byte{}, rights, nil)
+		conn.WriteMsgUnix([]byte{}, rights, socketAddr)
```

Comment:
I mostly think we should check the error because earlier versions of this code actually do have an error here (the "already connected socket" error!)

The @s-urbaniak version seems to work for me.  Might be slightly cleaner to put the custom code in a separate function but otherwise seems like the best idea, for now.  Maybe ping the thread too though?  One issue is that there's some sort of "freeze" since Go 1.4 on syscall and changes are supposed to happen on https://github.com/golang/sys/blob/master/unix/syscall_linux.go instead...